openmp/runtime/src/z_Linux_util.cpp

```diff
@@ -878,10 +881,19 @@ void __kmp_create_worker(int gtid, kmp_info_t *th, size_t stack_size) {
     KMP_SYSFAIL("pthread_create", status);
   }
 
-#ifdef LIBOMP_HAVE_LINUX_PTHREAD_SETNAME
+#if defined(LIBOMP_HAVE_PTHREAD_SETNAME_NP) ||                                 \
+    defined(LIBOMP_HAVE_PTHREAD_SET_NAME_NP)
   // Rename worker threads for improved debuggability
   if (!KMP_UBER_GTID(gtid)) {
+#if KMP_OS_LINUX
     pthread_setname_np(handle, "openmp_worker");
+#elif KMP_OS_FREEBSD || KMP_OS_OPENBSD || KMP_OS_DRAGONFLY
+    pthread_set_name_np(handle, "openmp_worker");
+#elif KMP_OS_NETBSD
+    pthread_setname_np(handle, "%s", const_cast<char *>("openmp_worker"));
+#elif KMP_OS_DARWIN
+    pthread_setname_np("openmp_worker");
```

Comment:
This sets the name of the current thread though? But what we want is to change the name of the newly created thread pointed to by `handle`?

Reply:
Good point. I took out the macOS handling.